management/commands/backfill_review_assignments.py

```diff
@@ -0,0 +1,102 @@
+from django.core.management.base import BaseCommand, CommandError
+
+import requests, json
+
+from journal.models import Journal
+from review.models import ReviewAssignment, ReviewForm, ReviewAssignmentAnswer
+from django.contrib.contenttypes.models import ContentType
+from utils.models import LogEntry
+from submission.models import Article
+
+from bs4 import BeautifulSoup
+
+class Command(BaseCommand):
+    """Backfills review assignment comments to a given journal"""
+    help = "Backfills review assignment comments to a given journal"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "journal_code", help="`code` of the journal to add arks", type=str
+        )
+        parser.add_argument(
+            "ojs_server", help="which server to read from dev/stg/prd", type=str
+        )
+
+    # Get the article OJS id from the import log
+    def get_ojs_id(self, article):
+        ctype = ContentType.objects.get(app_label='submission', model='article')
+        desc = f'Article {article.pk} imported by Journal Transporter.'
+        e = LogEntry.objects.filter(content_type=ctype, object_id=article.pk, description__startswith=desc)
+        if e.count() == 1:
+            d = json.loads(e[0].description.partition("Import metadata:")[2])
+            for i in d['external_identifiers']:
+                if i['name'] == "source_id":
+                    return i['value']
+
+        return None
+
+    # Extract id list from the OJS API
+    def get_ids(self, s):
+        return [x["source_record_key"].split(":")[-1] for x in json.loads(s)]
+
+    def strip_html(self, value):
+        soup = BeautifulSoup(value, 'html.parser')
+        return soup.get_text(separator="\n")
+
+    def handle(self, *args, **options):
+        code = options.get("journal_code")[:24]
+        server = options.get("ojs_server")
+
+        ojs_url = f"https://pub-submit2-{server}.escholarship.org/ojs/index.php/pages/jt/api/journals"
+
+        if not Journal.objects.filter(code=code).exists():
+            raise CommandError(f'Journal does not exist {code}')
+
+        journal = Journal.objects.get(code=code)
+        default_form = ReviewForm.objects.get(journal=journal, name="Default Form")
+        form_element = default_form.elements.all()[0]
+
+        # First, set the form to the default form for all Review Assignments
+        # that don't currently have one
+        x = ReviewAssignment.objects.filter(article__journal=journal, form=None)
+        for r in x:
+            r.form = default_form
+            r.save()
+
+        # Look for review comments for each article in this journal
+        for article in Article.objects.filter(journal=journal):
+            ojs_id = self.get_ojs_id(article)
+            round_url = f"{ojs_url}/{code}/articles/{ojs_id}/rounds"
+            rounds = self.get_ids(requests.get(round_url).text)
```

Comment:
we might consider some error handling here, if OJS isn't reachable for some reason? though this is just a simple script... I dunno, I can be convinced otherwise, but... it seems like some try/except blocks to up-armor the requests would be helpful